File: python/MP3_PartA.py
from pyspark import SparkContext
from pyspark.sql import SQLContext
from pyspark.sql.types import StructType
from pyspark.sql.types import StructField
from pyspark.sql.types import StringType, IntegerType

sc = SparkContext()
sqlContext = SQLContext(sc)

lines = sc.textFile("gbooks", 1)
parts = lines.map(lambda l: l.split("\t"))
records = parts.map(lambda p: {"word": p[0], "count1": int(p[1]), "count2": int(p[2]), "count3": int(p[3])})
schema = sqlContext.createDataFrame(records)
schema.registerTable("records")

####
# 1. Setup (10 points): Download the gbook file and write a function to load it in an RDD & DataFrame
####

# RDD API
# Columns:
# 0: place (string), 1: count1 (int), 2: count2 (int), 3: count3 (int)


# Spark SQL - DataFrame API

File: java/src/main/java/MP3_PartA.java
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;
import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.types.StructType;

public final class MP3_PartA {

  public static void main(String[] args) throws Exception {
    SparkSession spark = SparkSession
      .builder()
      .appName("MP3")
      .getOrCreate();


    try (JavaSparkContext sc = new JavaSparkContext(spark.sparkContext())) {
      StructType schema = new StructType()
        .add("word", "string", false)
        .add("count1", "integer", false)
        .add("count2", "integer", false)
        .add("count3", "integer", false);

      // SQLContext sqlContext = new SQLContext(sc);
      // Dataset<?> dataset = FileLoader.loadData(sc, sqlContext);
      Dataset<Row> dataset = spark.read().schema(schema).option("delimiter","\t").load("gbooks");
      dataset.createOrReplaceTempView("gbooks");
      dataset.printSchema();
      sc.stop();
    }

    // Finish up
    spark.stop();
  }
}

// Part A: Your_output_contains:_3_incorrect_lines_in_our_test.Please_try_again!      ;
// Part B: Congrats!_All_lines_are_correct!      ;
// Part C: Congrats!_All_lines_are_correct!      ;
// Part D: Congrats!_All_lines_are_correct!      ;
// Part E: Congrats!_All_lines_are_correct!      ;
